src/shared/clients/defaultEc2MetadataClient.ts

```diff
@@ -23,6 +23,18 @@ export class DefaultEc2MetadataClient implements Ec2MetadataClient {
         })
     }
 
+    getIamInfo(): Promise<IamInfo> {
+        return new Promise((resolve, reject) => {
+            this.metadata.request('/latest/meta-data/iam/info', (err, response) => {
+                if (err) {
+                    reject(err)
+                }
+                const iamInfo: IamInfo = JSON.parse(response)
```

Comment:
nit: could this and `getInstanceIdentity()` be abstracted to avoid the duplicated logic?

    return getThing<IamInfo>('/latest/meta-data/iam/info')